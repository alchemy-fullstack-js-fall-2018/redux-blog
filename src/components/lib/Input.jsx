import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledInput = styled.input`
  justify-self: start;
  margin: 5px;

  [type=text] {
    width: 100%;
  }
`;

const StyledLabel = styled.label`
  justify-self: end;
  margin: 5px;
`;

const Input = ({ onChange, resource, name }) => {

  const label = name.slice(0, 1).toUpperCase() + name.slice(1);

  return (
    <Fragment>
      <StyledLabel htmlFor={name}>{label}</StyledLabel>
      <StyledInput
        name={name} type="text"
        value={resource} onChange={onChange}
      ></StyledInput>
    </Fragment>
  );
};

Input.propTypes = {
  resource: PropTypes.any.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Input;
