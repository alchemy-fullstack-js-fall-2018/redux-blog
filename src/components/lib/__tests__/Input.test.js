import React from 'react';
import { shallow } from 'enzyme';
import Input from '../Input.jsx';

describe('<Input />', () => {
  test('renders', () => {
    const onChange = jest.fn();
    const resource = 'abc';
    const name = 'xyz';
    const wrapper = shallow(
      <Input
        resource={resource}
        name={name}
        onChange={onChange}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
