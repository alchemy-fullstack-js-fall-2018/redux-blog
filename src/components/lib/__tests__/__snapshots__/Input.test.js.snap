// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders 1`] = `
<Fragment>
  <styled.label
    htmlFor="xyz"
  >
    Xyz
  </styled.label>
  <styled.input
    name="xyz"
    onChange={[MockFunction]}
    type="text"
    value="abc"
  />
</Fragment>
`;
